LiteCore/Query/QueryParser.cc

```diff
@@ -286,9 +286,9 @@ namespace litecore {
     }
 
 
-    void QueryParser::writeCreateIndex(const Array *expressions) {
+    void QueryParser::writeCreateIndex(const string &name, const Array *expressions) {
         reset();
-        _sql << "CREATE INDEX IF NOT EXISTS \"" << indexName(expressions) << "\" ON " << _tableName << " ";
+        _sql << "CREATE INDEX IF NOT EXISTS \"" << name << "\" ON " << _tableName << " ";
```

Comment:
Should take out `IF NOT EXISTS` because if there's already an index with that name, it's no longer a no-op but an error (since the index expression might be different.)